lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -415,6 +415,11 @@ static void collectPossibleCalleesByQualifiedLookup(
   auto baseInstanceTy = baseTy->getMetatypeInstanceType();
   if (!baseInstanceTy->mayHaveMembers())
     return;
+  auto *baseNominal = baseInstanceTy->getAnyNominal();
+  if (baseNominal) {
+    // Force all implicit members synthesized before lookup.
+    (void)baseNominal->getAllMembers();
+  }
```

Comment:
I found it a bit odd this wasn't already handled in the lookup. Is there a bug there? Ie. it does call `NominalTypeDecl::prepareLookupTable`. Is it intentional that calls `getMembers` rather than `getAlllMembers`? `getMembers` calls `loadAllMembers` but that doesn't get synthesized implicit members... How confusing

Reply:
I agree this is odd. I believe what is happening is that when LookupVisibleDecls does a walk over the members of a nominal it uses getMembers() and not getAllMembers(). Perhaps it should use getAllMembers() if we wish to see all possible synthesized declarations, and not just those that have been synthesized up until now.

Rintaro, do you mind investigating and pushing this down into the appropriate part of LookupVisibleDecls?

Reply:
@slavapestov  FWIW, this is not `LookupVisibleDecls`, but `DeclContext::lookupQualified()`.

But before I opened this PR,  I tested `getAllMembers()` in LookupVisibleDecls and [eliminating our own logic force triggering synthesizing implicit members](https://github.com/apple/swift/blob/e737770cdb5fcb5979111365a727ca4d2d33dc2c/lib/Sema/LookupVisibleDecls.cpp#L572-L599). In general, that works, but it caused a circular dependency error.
```
struct S {
  var value = True
}
```
When resolving `True`, lookup fails and tries to perform type correction (in test cases which we still do typo correction). When typo correction lookups visible decls, StructDecl("S")->getAllMembers() tries to return `value` and its type, which invokes interface type request on `value`, which tries to resolve `True`. Do you have a good idea to resolve this circular dependency?

Reply:
@bnbarham  We talked about this offline, but qualified lookup (`prepareLookupTable`) uses `getMembers()` because it doesn't want synthesize all implicit members. It wants only members with the specified name. `TypeChecker::lookupMember` has the trick [`installSemanticMembersIfNeeded()` ](https://github.com/apple/swift/blob/e737770cdb5fcb5979111365a727ca4d2d33dc2c/lib/Sema/TypeCheckNameLookup.cpp#L276-L281) which trigger the synthesis of those named members. Probably we should do similar things here instead of calling `getAllMembers()` which does unnecessary works.

Or, maybe we should pursue sinking this logic inside `DeclContext::lookupQualified()`

Reply:
Yeah, re-using installSemanticMembersIfNeeded() sounds like the correct approach. Sinking the logic down into DeclContext::lookupQualified() will probably take a lot of work to fix all the cycles.